### ISSUE TITLE
Unable to open ZIM files on Samsung device with Android 12

### ISSUE THREAD

**Issue by Caleb13-2** [kiwix/kiwix-android#2779]
I know that similar bugs had been reported in the past, but I think I encountered a new variant. I recently bought Samsung Galaxy Tab S7+ (part number SM-T970) and I upgraded it to Android 12 (core 4.19.113-2322000). Later, I installed Kiwix 3.4.5 via APK file, but I'm unable to open any ZIM files. It dosn't matter if they're in the tablet's internal Flash or on SD card. (Files downloaded via Kiwix inferface work fine, though). The tablet comes with a file manager app named "My Files" but it's next to useless - when I click on a ZIM file and select "open with", it redirects me to Play Store where it finds everything that contains "zim" string. So I installed Total Commander, but it didn't help either. The "open with" option it TC actually allows me to select Kiwix, but the ZIM file doesn't load. Moreover, Kiwix's behavior depends on whether it was already running or not. Steps to reproduce it:

Scenario A:
1. I close all running aps.
2. I run Total Commander and switch to directory with the ZIM files.
3. I tap on one ZIM file and select "open with", then I click on Kiwix icon.
4. Kiwix starts up to the "no open book" screen, but the ZIM file doesn't load.

Scenario B:
1. I do the 4 steps as above.
2. Then I switch back to Total Commander and try to open the same ZIM file again.
3. Kiwix comes to foreground, but this time a really odd error message appears, "Please try looking for this file in the Device Tab of your Library." See the screeshot I attached.

I can provide you with logs etc., but I'm not an Android fan, so please provide me with step-by-step instructions how to obtain them.

![Screenshot_20220125-222119_Kiwix](https://user-images.githubusercontent.com/70209677/151068492-f367fc70-807b-4559-a531-5daf36e09389.jpg)

**Comment by ru2saig** [comment]
Could you try the development version here: http://download.kiwix.org/nightly/2022-02-08/dev-universal-debug.apk? I had the same issue you had, but this debug version fixed it. I think this will be for the next stable release 3.5.6.

Warning: the screen will flash on screen updates!

**Comment by Caleb13-2** [comment]
Yes, it works fine with the development version. Kiwix even automatically found all the ZIM files on the SD card. Thanks.
